DOC-4148: Update Data structure pages (#59)
exoplatform/exo-documentation: docs/Platform_Development.rst

   writes activities in the space activity stream when events or tasks
   are added/modified.

.. _PLFRefGuide.PLFDevelopment.DataStructure:

================
Data Structure
================

This section consists of the following main topics:

-  :ref:`Social Data structure <PLFRefGuide.PLFDevelopment.DataStructure.Social>`

   A comprehensive knowledge of the Social Data Structure that is organized to 

>    conform to the data storage for the individual purpose of Social.

Reviewer: I am not sure to understand the sentence. It should be simpler IMO, something like "A description of the Social Data Structure".

Author: OK

   calendarSetting.
   
-  :ref:`Wiki Data structure <PLFRefGuide.PLFDevelopment.DataStructure.Wiki>`

   Introduction to the whole Data structure of Wiki, and comprehensive 
   knowledge of its main nodes: Wiki data and Wikimetadata.   
   
-  :ref:`Forum JCR structure <PLFRefGuide.PLFDevelopment.DataStructure.Forum>`

   Introduction to the whole JCR structure of Forum, and comprehensive
   knowledge of its main nodes: Forum System and Forum Data.

-  :ref:`FAQ JCR structure <PLFRefGuide.PLFDevelopment.DataStructure.FAQ>`

   Introduction to the whole JCR structure of FAQ, and comprehensive
   knowledge of its main nodes: Category, FAQ setting, Template for FAQ.

-  :ref:`Poll JCR structure <PLFRefGuide.PLFDevelopment.DataStructure.Poll>`

   Introduction to the whole JCR structure of Poll, and properties of
   its node type (exo:polls).

.. note:: To learn more about the JCR Structure, you should have the certain knowledge of `JCR <http://jcp.org/en/jsr/detail?id=170>`__.

.. _PLFRefGuide.PLFDevelopment.DataStructure.Social:

Social Data structure
~~~~~~~~~~~~~~~~~~~~~~

Social data is stored in the :ref:`JPA data source <Database.ConfiguringPLF>`

> in a set of database tables as the following:

Reviewer: "as follows" ?

|image13|

> The table **SOC\_ACTIVITIES** stores all the activities and comments and has these attributes:

Reviewer: ```suggestion
The table **SOC\_ACTIVITIES** stores all the activities and comments and has these fields:
```

+-----------------------+-----------------+------------------------------------------------------------------------------------+

> | Property name         | Required type   | Description                                                                        |

Reviewer: ```suggestion
| Name         | Type   | Description                                                                        |
```

+=======================+=================+====================================================================================+

> | ACTIVITY_ID           | ID              | The unique ID of the activity.                                                     |

Reviewer: ```suggestion
| ACTIVITY_ID           | ID              | The unique ID of the activity or comment.                                                     |
```

node: **CalendarApplication/calendars/%calendar\_id%**. Its node type is
**exo:calendar** that has the following properties:

+-----------------------+-----------------+------------------------------------------------------------------------------------------------------------------------------------------------------+

> | Field                 |          type   |  Description                                                                                                                                         |

Reviewer: The calendar application is still in JCR, so we should not update this table IMO.

Author: I did not change it, it looks like a bad copy paste of the table heading

Reviewer: ok so you will revert it ?
It looks like a column has been removed also (Multiple).

Author: The table structure was affected so it appears that the whole table was removed then added again (you notice this when you open whole file)

Author: It should be OK now

+-----------------------+-----------------+------------------------------------------------------------------------------------+
| DRAFT_PAGE_ID         | ID              | The unique ID of the wiki draft page attachment.                                   |
+-----------------------+-----------------+------------------------------------------------------------------------------------+
| CREATED_DATE          | TIMESTAMP       | The creation date of the wiki draft page attachment.                               |
+-----------------------+-----------------+------------------------------------------------------------------------------------+
| FULL_TITLE            | VARCHAR         | The title of the wiki draft page attachment.                                       |
+-----------------------+-----------------+------------------------------------------------------------------------------------+
| ATTACHMENT_FILE_ID    | ID              | The unique ID of the file in the table FILES_FILES.                                |
+-----------------------+-----------------+------------------------------------------------------------------------------------+

> Table **WIKI\_WATCHERS*

Reviewer: ```suggestion
Table **WIKI\_WATCHERS**
```

Author: (y) thank you!

| exo:activityId     | String          | false      | When a Poll is created, a new activity will be created. The activity Id is stored as preference between Poll and the activity. When there are any updates on the poll, the corresponding activities will be updated on the Activity Stream.   |
+--------------------+-----------------+------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

.. _PLFRefGuide.PLFDevelopment.DataStructure.LoginHistory:

Login History data structure
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The login history data is stored in this database table:

|image19|

Table **LOGIN\_HISTORY**

+-----------------------+-----------------+------------------------------------------------------------------------------------+
| Name                  | Type            | Description                                                                        |
+=======================+=================+====================================================================================+

> | ID                    | ID              | The unique ID of the login history.                                                |

Reviewer: ```suggestion
| ID                    | ID              | The unique ID of the login history entry.                                                |
```

+-----------------------+-----------------+------------------------------------------------------------------------------------+

> | USER_ID               | VARCHAR         | The ID of related user to the login history.                                       |

Reviewer: ```suggestion
| USER_ID               | VARCHAR         | The ID of related user to the login history entry.                                       |
```

+-----------------------+-----------------+------------------------------------------------------------------------------------+

> | LOGIN_DATE            | TIMESTAMP       | The time of the last user's login date.                                            |

Reviewer: ```suggestion
| LOGIN_DATE            | TIMESTAMP       | The date of the login history entry                                            |
```
